### DISCUSSION TITLE
ElevenLabs TTS HUngarian speach

### DISCUSSION THREAD

**Discussion by laszlolaszlo** [SubtitleEdit/subtitleedit#8810] [Q&A] [answered]
Hello,

I love SE! 
I like to use ElevenLabs TTS in Hungarian mode. On ElevelLabs WebUI I am able to use with Eleven Turbo v2.5 modell. It contains Hungarian language support.
I updated the voice list using my API key. The Hungarian voice I recorded on the WebUI also appeared. Despite this, the generated test tried to pronounce the Hungarian text in English. Is it possible that when making the API call, we also need to specify the language for ElevenLabs to use?
![image](https://github.com/user-attachments/assets/a74179e9-fbeb-4e3d-b3b5-dca3d8763007)

Can you check this?
Thank you

**Comment by laszlolaszlo** [comment]
With model_id="eleven_turbo_v2_5" you should use `language_code` parameter.
https://elevenlabs.io/docs/api-reference/text-to-speech

I tried with a Python script, it is just works fine. So if you have some time please implement this, Thank you so much!

**Comment by niksedk** [comment]
I can give it a try but the documentation sucks... what should language code be used for "English (US)"?

**Reply by laszlolaszlo** [reply]
Thank you so much! It should be greate!
Language codes are here:
https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes

ElevenLabs need ISO 639-1 language code for `language_code `parameter.

**Reply by niksedk** [reply]
He, this I know... 

Elevenlabs list "English US" + "English GB" + "English CA" languages... but `ISO 639-1` only has `en` !?

**Reply by laszlolaszlo** [reply]
Ohhh... I see. 
https://elevenlabs.io/docs/api-reference/text-to-speech
There is ony English so I think you can not fine tune English.
https://help.elevenlabs.io/hc/en-us/articles/19581255545873-How-do-I-select-the-language-and-accent

**Comment by niksedk** [comment]
How is this? https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.8/SubtitleEditBeta.zip

Still not sure what language code to use...

**Reply by laszlolaszlo** [reply]
Looks good for Hungarian language!
I can use my own voice (Matt) too. Thank yoz so much!
![image](https://github.com/user-attachments/assets/5793c293-6e2b-41be-ba7e-a7aa1483d5db)

**Reply by niksedk** [reply]
New beta upped: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.8/SubtitleEditBeta.zip
Now only ISO 639-1 language codes.

Let me know how it works for you. 
And all voices works with both models?

**Reply by laszlolaszlo** [reply]
Thank you! It caused a Null Pointer Exception at Adjust Speed. I tried two times.

![image](https://github.com/user-attachments/assets/f92ad899-6614-473b-b68d-08f6674a921e)

**Reply by niksedk** [reply]
I cannot re-create this error...
Can you email or attach the `error_log.txt` file?

**Comment by laszlolaszlo** [comment]
Here it is:
[error_log.txt](https://github.com/user-attachments/files/16959174/error_log.txt)

**Reply by niksedk** [reply, marked as answer]
OK, hopefully fixed the divide-by-zero now: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.8/SubtitleEditBeta.zip

Could some of lines in your subtitle have a duration of zero?
Perhaps SE should do some validating...

**Reply by laszlolaszlo** [reply]
Looks good! Thank you so much!
The subtitle did not contains 0 duration.
